Build Docker image on slim base

AndrewFerr opened the pull request:
This reduces the size of the runtime image by about 50%.

depends on https://github.com/element-hq/element-call/pull/3294

AndrewFerr committed:
Build Docker image on slim base

toger5 commented:
Can we track what tests we did with the slim-base to make sure its not breaking anything.
 - [ ] deploy and run EC from the slim-base version

AndrewFerr committed:
Merge branch 'livekit' into af/docker/slim

AndrewFerr committed:
Add missing EW volume label for playwright compose

AndrewFerr committed:
Run Playwright tests against Docker container

For Playwright end-to-end tests in CI, instead of running a development
webserver with `yarn dev`, build and deploy a Docker container for
Element Call and use that as the webserver to test against.

AndrewFerr commented:
> Can we track what tests we did with the slim-base to make sure its not breaking anything.

There weren't any existing tests that used Docker, so I added 073baca to run the Playwright tests against a freshly-built Docker container of EC.  This should ensure that not only does the Docker image build properly, but that running containers can successfully serve EC.

AndrewFerr commented on playwright/fixtures/widget-user.ts:
```diff
@@ -63,16 +63,27 @@ const CONFIG_JSON = {
+          "Developer.elementCallUrl",
+          null,
+          "device",
+          "https://localhost:3000/room",
```
For some reason, this wasn't able to use a variable, so I've defined two different versions of the function for each value of the URL.

AndrewFerr committed:
Try to fix CI

AndrewFerr commented:
Not sure why the end-to-end tests are hanging in CI, and they run properly for me in an Ubuntu VM...

toger5 commented:
@AndrewFerr they are currently broken.
We need to get them fixed before merging (We did a couple of exceptions for the release relevant PR's)
We will merge it in your name once we have them fixed

AndrewFerr committed:
Merge branch 'livekit' into af/docker/slim

AndrewFerr commented:
I noticed "playwright" in some recent commit messages, so here's a merge to try tests again :slightly_smiling_face: 

EDIT: Still no luck.

AndrewFerr committed:
Merge branch 'livekit' into af/docker/slim

fkwp committed:
Merge branch 'livekit' into af/docker/slim

fkwp committed:
Merge branch 'fkwp/docker_compose/resolve_conflicting_ports' into af/docker/slim

fkwp committed:
Merge branch 'fkwp/docker_compose/resolve_conflicting_ports' into af/docker/slim

fkwp committed:
fix USE_DOCKER port

fkwp committed:
fix element web fixture protocol

fkwp committed:
Merge branch 'fkwp/docker_compose/resolve_conflicting_ports' into af/docker/slim

fkwp committed:
wip

fkwp committed:
Merge branch 'livekit' into af/docker/slim

AndrewFerr committed:
Build test image to local Docker registry

AndrewFerr committed:
Shut down playwright webserver gracefully

When using a containerized webserver, this stops the container once
tests finish.

AndrewFerr committed:
Use full test image name when running it

AndrewFerr committed:
Enable verbose Playwright logging in CI

AndrewFerr committed:
Increase Playwright timeout in CI

AndrewFerr commented:
Now we're getting somewhere.  Increasing the test job timeout revealed [the errors that the tests are hitting](https://github.com/element-hq/element-call/actions/runs/15470239945/job/43552420476?pr=3204#step:8:2142), which do not reproduce when running them in a local Ubuntu VM.

The tests are failing on expected page elements not being visible.  Any idea what could be causing that?

AndrewFerr committed:
Revert "Enable verbose Playwright logging in CI"

This reverts commit 298aced3c35f491d446618d22dc208cc1e361366.
The point of this was to figure out why the tests were being cancelled.

AndrewFerr committed:
Upload nginx logs for Playwright CI test

AndrewFerr committed:
Fix typo in Docker volume location

fkwp reviewed (changes requested):
Nice catch https://github.com/element-hq/element-call/pull/3204/commits/5a849360131b9efd764c405f2124dbcb77878171

I think to get it over the line removing the debugging logs to track down https://github.com/element-hq/element-call/pull/3204/commits/5a849360131b9efd764c405f2124dbcb77878171 would be nice.

fkwp commented on scripts/playwright-webserver-command.sh:
```diff
@@ -0,0 +1,13 @@
+if [ -n "$USE_DOCKER" ]; then
+    set -ex
+    yarn build
+    IMAGE_NAME=localhost:5000/element-call:testing
```
What is the purpose of localhost:5000 here as `IMAGE_NAME` prefix?

AndrewFerr replied:
This was to explicitly use the local image registry for the built image, just in case `docker run element-call:testing`  defaulted to pulling from docker.io.  It doesn't, so this can be removed.

AndrewFerr replied:
Reverted in d8ec2e6

AndrewFerr committed:
Revert debugging changes for Playwright

AndrewFerr committed:
Fix new typo
